### PR TITLE
HDDS-7816. Add DataNode list to the SCM WebUI

### PR DESCRIPTION
**Pull request by debiswal**
**What changes were proposed in this pull request?**
commit id : 579d1eabdd565269fb3830e0065646d1b77f4cf9 (holding all backend & UI changes)
This Includes Sorting, Searching & Pagination on SCM UI where page deafult size is 10. we can also set this to 20 or 50 records per page as well. 
Page slicing & Record Slicing  both have been implemented along wth sort & search.

**Backend Changes** ::    
JMX API developed  & the response from API is 

"NodeStatusInfo" : [ {
      "key" : "ozone_datanode_1.ozone_default",
      "value" : [ "HEALTHY", "IN_SERVICE" ]
    }, {
      "key" : "ozone_datanode_3.ozone_default",
      "value" : [ "HEALTHY", "IN_SERVICE" ]
    }, {
      "key" : "ozone_datanode_2.ozone_default",
      "value" : [ "HEALTHY", "IN_SERVICE" ]
    } ]
	

**Note**: HDDS 7816 JIRA includes both UI & Backend changes .

**What is the link to the Apache JIRA**
https://issues.apache.org/jira/browse/HDDS-7816

**How was this patch tested?**

   Manually 
Attached screenshot for reference

image

**Search**: 
image

**Sort**: 
image

### CODE REVIEW

**Comment by adoroszlai** [comment]
@debiswal Thanks for the patch.

Please [request an account](https://infra.apache.org/jira-guidelines.html#who) for [Apache Jira](https://issues.apache.org/jira/) (project you need to substitute in the email address is `ozone`).  When your Jira account is created, please file an [HDDS issue](https://issues.apache.org/jira/browse/HDDS) and link it in the PR description.  Also update the PR title to include the Jira issue ID in `HDDS-1234. ` format as a prefix.

Also, please include in the PR description information on how the change was tested.  Since it's a UI change, I assume it was tested manually.  Please include some screenshot for reference.

File: hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm.js [hunk 1]
```diff
@@ -24,12 +24,46 @@
...
+            let nodeStatusCopy=[];
+            $scope.RecordsToDisplay ="10";
+            $scope.currentPage=1;
+            $scope.lastIndex = 0;
+
             $http.get("jmx?qry=Hadoop:service=SCMNodeManager,name=SCMNodeManagerInfo")
                 .then(function (result) {
                     ctrl.nodemanagermetrics = result.data.beans[0];
-                });
+                $scope.nodeStatus = ctrl.nodemanagermetrics && ctrl.nodemanagermetrics.NodeStatus && ctrl.nodemanagermetrics.NodeStatus.map(({ key, value}) =>{
```
Comment (fapifta) [inline hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm.js:39]: From this point on indentation seems to be shifted and a bit ad-hoc, please fix that, and also please go over the file, and fix missing spaces here and there (around = sign, after { and before }, after ,)

**Review by fapifta — COMMENTED** [review]
Thank you for working on this patch @debiswal , I have added some additional comments to the already present code inline, please take care of those as well.

File: hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html [hunk 2]
```diff
@@ -28,25 +28,53 @@ <h2>SCM Information</h2>
...
-
-<table class="table table-bordered table-striped" class="col-md-6">
+<h2>Node Status</h2>
+<div class="row">
+    <div class="col-md-6 text-left">
+        <label>Show: </label>
+        <select class="form-select" ng-model="RecordsToDisplay" ng-change="UpdateRecordsToShow()">
+            <option value="10" ng-selected="{true}">10</option>
+            <option value="20">20</option>
+            <option value="50">50</option>
```
[the remark this answers is not in the document]
Comment (debiswal) [reply hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html:38]: commit id : b99acf9c78e7b8b7cbdb64d609aa14bf8fcad8e9 
this commit includes option 'All' added in SCM UI for user.

File: hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm.js (hunk 1 above)
Comment (debiswal) [reply hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm.js]: Indentation has been fixed as well.

**Comment by adoroszlai** [comment]
@debiswal This PR has unrelated commits from other contributors.  If you want to update your branch, please merge `master` into it instead of cherry-picking individual commits.

**Comment by debiswal** [comment]
Closing this PR due to  unrelated commits as new PR has been raised . 
**New PR Link :** 
https://github.com/apache/ozone/pull/4289

**Pull request closed without merging**